app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -74,8 +74,44 @@ def boot_from_volume(args = {})
       } ]
     end
 
+    def create_additional_volume_api(args)
+      vm_name = args[:name]
+
+      # Need to specify image ref on boot_index 0
+      args[:block_device_mapping_v2] = [ {
+        :source_type => "image",
+        :delete_on_termination =>  true,
+        :boot_index => 0,
+        :uuid => args[:image_ref],
+        :destination_type => "local"
+      } ]
+
+      # Iterate into volumes attributes
+      args[:volumes_attributes].each do |key, value|
+        boot_index = key.to_i + 1
```

Comment:
This sets a 'hidden contract' where the order of the volumes you specify determines the boot order. Can you make this use boot_index from the arguments? If it's not available, then fine, default to this. Leave a comment mentioning it, as we would have to add that to the documentation.